api/testing.py

```diff
@@ -25,87 +26,74 @@
     temp_config,
 )
+
 from api.adobe_vendor_id import AuthdataUtility
 
 class VendorIDTest(DatabaseTest):
-    """Sets up an Adobe Vendor ID integration."""
+    """A DatabaseTest that knows how to set up an Adobe Vendor ID
+    integration.
+    """
 
-    TEST_NODE_VALUE = 114740953091845
     TEST_VENDOR_ID = u"vendor id"
-    TEST_LIBRARY_URI = u"http://me/"
-    TEST_OTHER_LIBRARY_URI = u"http://you/"
-    TEST_OTHER_LIBRARIES  = {TEST_OTHER_LIBRARY_URI: ("you", "secret2")}
-
-    TEST_SHORT_NAME = u'LBRY'
-    TEST_SHARED_SECRET = u'some secret'
-
-    def setup(self, _db=None):
-        super(VendorIDTest, self).setup()
-
-        if not _db:
-            # So long as we're not testing a scoped session, create
-            # the Adobe Vendor ID and Library Registry credentials.
-            self.adobe_vendor_id = self._external_integration(
-                ExternalIntegration.ADOBE_VENDOR_ID,
-                ExternalIntegration.DRM_GOAL, username=self.TEST_VENDOR_ID)
-            self.set_main_library_adobe_config(self._default_library)
-
-            self.short_client_token = self._external_integration(
+    
+    def initialize_adobe(self, vendor_id_library, short_token_libraries=[]):
+        """Initialize an Adobe Vendor ID integration and a number of
+        Short Client Token integrations.
+
+        :param vendor_id_library: The Library that should have an
+        Adobe Vendor ID integration.
+
+        :param short_token_libraries: The Libraries that should have a
+        Short Client Token integration.
+        """
+        if not vendor_id_library in short_token_libraries:
+            short_token_libraries.append(vendor_id_library)
+        # The first library acts as an Adobe Vendor ID server.
+        self.adobe_vendor_id = self._external_integration(
+            ExternalIntegration.ADOBE_VENDOR_ID,
+            ExternalIntegration.DRM_GOAL, username=self.TEST_VENDOR_ID,
+            libraries=[vendor_id_library]
+        )
+            
+        other_libraries = dict()
+            
+        # Every library in the system can generate Short Client
+        # Tokens.
+        for library in short_token_libraries:
+            short_client_token = self._external_integration(
                 ExternalIntegration.SHORT_CLIENT_TOKEN,
                 ExternalIntegration.DRM_GOAL,
-                libraries=[self._default_library]
+                libraries=[library]
             )
-            self.set_main_short_client_token_configuration(self._default_library)
 
+            # Each library will get a slightly different short
+            # name and secret for generating Short Client Tokens.
+            library_uri = self._url
+            short_name = library.short_name + "token"
+            secret = library.short_name + " token secret"
+            short_client_token.username = short_name
+            short_client_token.password = secret
+
+            library.setting(Library.WEBSITE_KEY).value = library_uri
```

Comment:
A library uses the URL of its website (e.g. https://nypl.org/) to generate JWTs. This URL is shared with the library registry -- it's the keys of the dictionary in the 'other_libraries' setting.

This URL is also going to be an important part of the Authentication for OPDS document, so Courteney and I decided it made more sense as a library-wide setting than as part of the Adobe Vendor ID and Short Client Token integrations.